appveyor-linux-binary.yml

```diff
@@ -0,0 +1,296 @@
+  - sh: "source $HOME/.cargo/env"
+  - sh: "rustup toolchain install stable --profile minimal --no-self-update"
+  - sh: "rustup default stable"
+  - sh: "rustup target add x86_64-unknown-linux-gnu --toolchain stable"
+  - sh: "rustup target add aarch64-unknown-linux-gnu --toolchain stable"
+  - sh: "pip install cargo-lambda==$CARGO_LAMBDA_VERSION"
+  - sh: "rustc -V"
+  - sh: "cargo -V"
+  - sh: "cargo lambda -V"
+
+  - sh: "python3.9 -m venv .venv_env_vars"
+  - sh: ".venv_env_vars/bin/pip install boto3"
+  - sh: "test_env_var=$(.venv_env_vars/bin/python tests/get_testing_resources.py)"
+  - sh: '
+    if [ $? -ne 0 ]; then
+      echo "get_testing_resources failed. Failed to acquire credentials or test resources.";
+      false;
+    fi
+  '
+
+  - sh: 'export CI_ACCESS_ROLE_AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID'
+  - sh: 'export CI_ACCESS_ROLE_AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY'
+  - sh: 'export CI_ACCESS_ROLE_AWS_SESSION_TOKEN=$AWS_SESSION_TOKEN'
+
+  - sh: 'export AWS_ACCESS_KEY_ID=$(echo "$test_env_var" | jq -j ".accessKeyID")'
+  - sh: 'export AWS_SECRET_ACCESS_KEY=$(echo "$test_env_var" | jq -j ".secretAccessKey")'
+  - sh: 'export AWS_SESSION_TOKEN=$(echo "$test_env_var" | jq -j ".sessionToken")'
+  - sh: 'export TASK_TOKEN=$(echo "$test_env_var" | jq -j ".taskToken")'
+  - sh: 'export AWS_S3_TESTING=$(echo "$test_env_var" | jq -j ".TestBucketName")'
+  - sh: 'export AWS_ECR_TESTING=$(echo "$test_env_var" | jq -j ".TestECRURI")'
+  - sh: 'export AWS_KMS_KEY=$(echo "$test_env_var" | jq -j ".TestKMSKeyArn")'
+  - sh: 'export AWS_SIGNING_PROFILE_NAME=$(echo "$test_env_var" | jq -j ".TestSigningProfileName")'
+  - sh: 'export AWS_SIGNING_PROFILE_VERSION_ARN=$(echo "$test_env_var" | jq -j ".TestSigningProfileARN")'
+
+  # required for RIE with arm64 in linux
+  - sh: "
+    if [[ -n $BY_CANARY ]] && [[ -n $DOCKER_USER ]] && [[ -n $DOCKER_PASS ]];
+      then echo Logging in Docker Hub; echo $DOCKER_PASS | docker login --username $DOCKER_USER --password-stdin registry-1.docker.io;
+    fi"
+  - sh: "
+    if [[ -n $BY_CANARY ]] && [[ -n $DOCKER_USER ]] && [[ -n $DOCKER_PASS ]];
+      then echo Logging in Docker Hub; echo $DOCKER_PASS | docker login --username $DOCKER_USER --password-stdin;
+    fi"
+  - sh: "docker run --rm --privileged multiarch/qemu-user-static --reset -p yes"
+
+
+  # Runs only in Linux, logging Public ECR when running canary and cred is available
+  - sh: "
+    if [[ -n $BY_CANARY ]];
+      then echo Logging in Public ECR; aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws;
+    fi"
+
+
+build_script:
+  - curl -L https://github.com/aws/aws-sam-cli/releases/download/sam-cli-nightly/aws-sam-cli-linux-x86_64.zip -o aws-sam-cli-linux-x86_64.zip
+  - unzip aws-sam-cli-linux-x86_64.zip -d sam-installation
+  - sudo ./sam-installation/install
+  - sudo mv /usr/local/bin/sam-nightly /usr/local/bin/sam
```

Comment:
nit: worth adding some comment that we are overriding executable name (`sam-nightly`) with `sam` here.